src/OgMembershipInterface.php

```diff
@@ -49,4 +54,186 @@
    */
   const REQUEST_FIELD = 'og_membership_request';
 
+  /**
+   * Gets the membership creation timestamp.
+   *
+   * @return int
+   *   The membership creation timestamp.
+   */
+  public function getCreatedTime();
+
+  /**
+   * Sets the membership creation timestamp.
+   *
+   * @param int $timestamp
+   *   The membership creation timestamp
+   *
+   * @return OgMembershipInterface
+   */
+  public function setCreatedTime($timestamp);
+
+  /**
+   * Sets the membership's owner.
+   *
+   * @param mixed $etid
+   *   The user's ID or object.
+   *
+   * @return OgMembershipInterface
+   */
+  public function setUser($etid);
+
+  /**
+   * Gets the membership's owner.
+   * 
+   * @return User
+   *   The user object.
+   */
+  public function getUser();
+
+  /**
+   * Sets the membership field name.
+   *
+   * A user can have two group reference fields. The field name property helps
+   * us to know to which field the membership belongs.
+   *
+   * @param string $fieldName
+   *   The group reference field name.
+   *
+   * @return OgMembershipInterface
+   */
+  public function setFieldName($fieldName);
+
+  /**
+   * Gets the membership field name.
+   *
+   * @return string
+   *   The group reference field name.
+   */
+  public function getFieldName();
+
+  /**
+   * Sets the group entity type ID.
+   *
+   * @param mixed $gid
+   *   The group entity type ID or name.
+   *
+   * @return OgMembershipInterface
+   */
+  public function setEntityId($gid);
+
+  /**
+   * Gets the group entity type ID.
+   *
+   * @return integer
+   *   The entity identifier.
+   */
+  public function getEntityId();
+
+  /**
+   * Sets the group entity type ID.
+   *
+   * @param mixed $groupType
+   *   The group entity type ID or object.
+   *
+   * @return OgMembershipInterface
+   */
+  public function setEntityType($groupType);
+
+  /**
+   * Gets the group entity type ID.
+   *
+   * @return string
+   *   The group entity type ID.
+   */
+  public function getGroupEntityType();
```

Comment:
To be consistent with `setEntityType()` I would rename this to `getEntityType()`.

I think it is also more the convention to use "Returns the [thing]" instead of "Gets the [thing]", but that's very minor, it's perfectly clear.